ansible_collections/arista/avd/plugins/plugin_utils/schema/avdvalidator.py

```diff
@@ -155,6 +167,27 @@ def _valid_values_validator(validator, valid_values, instance, schema: dict):
         yield jsonschema.ValidationError(f"'{instance}' is not one of {valid_values}")
 
 
+def _password_validator(validator, password_type: str, password_key: str, instance: dict, schema: dict):
+    """
+    This function validates if password is a valid password for the given password_type
+
+    The schema validates that the password_type is a valid one so no check is run on this method
+    """
+    password = instance["password"]
```

Comment:
Please look into my suggestion of moving the logic. This line is effectively hardcoding the key to `password` even if the schema key is something else.